disk/src/com/treode/disk/LogIterator.scala

```diff
@@ -50,6 +51,8 @@ private class LogIterator private (
   private var logPos = superb.logHead
   private var pagePos = Option.empty [Long]
   private var pageLedger = new PageLedger
+  var checkpointer = kit.checkpointer;
```

Comment:
The iterator never changes these, so they could be vals rather than vars.

    var checkpointer = kit.checkpointer;

However, in Scala there's an even better way:

    import kit.{checkpointer, compactor}

You'll see some imports like that just a few lines above.